openrtb/macrosubs_test.go

```diff
@@ -6,106 +6,98 @@ import (
-		t.Fatal("MacroSubs should complete successfully")
-	}
+type auctionResult struct {
+	price float64
+}
 
-	if result != expectedOutput {
-		t.Errorf("Got the wrong result.\nExpected: %s\nActual: %s", expectedOutput, result)
-	}
+func (ar *auctionResult) Price() float64 {
+	return ar.price
 }
 
-// TestMacroSubs creates a number of tests cases and passes them to verifyMacroSubs.
 func TestMacroSubs(t *testing.T) {
-	var bid = openrtb.Bid{
+	bid := openrtb.Bid{
 		AdId:         "Some ad id goes here",
 		Id:           "TheBidId!",
 		ImpressionId: "ImpressionIdForBid",
 		Price:        2.345,
 	}
-	var seatbid = openrtb.SeatBid{
+	seatBid := openrtb.SeatBid{
 		Seat: "SeatBidIdentifier",
 		Bids: []*openrtb.Bid{&bid},
 	}
-	var bidRes = &openrtb.BidResponse{
+	bidRes := &openrtb.BidResponse{
 		Id:       "1234",
-		SeatBids: []*openrtb.SeatBid{&seatbid},
+		SeatBids: []*openrtb.SeatBid{&seatBid},
 		Currency: openrtb.CURRENCY_USD,
 	}
-
-	var macroSubsTests = []struct {
-		input          string
-		bidRes         *openrtb.BidResponse
-		expectedOutput string
+	ar := &auctionResult{price: 2.345}
+	tests := []struct {
+		input    string
+		expected string
 	}{
-		{"abc${AUCTION_ID}def", bidRes, "abc1234def"},
-		{"abc${AUCTION_IDD}def", bidRes, "abc${AUCTION_IDD}def"},
-		{"abc${AUCTION_ID:B64}def", bidRes, "abcMTIzNA==def"},
-		{"abc${AUCTION_BID_ID}def", bidRes, "abcTheBidId!def"},
-		{"abc${AUCTION_IMP_ID}def", bidRes, "abcImpressionIdForBiddef"},
-		{"abc${AUCTION_SEAT_ID}def", bidRes, "abcSeatBidIdentifierdef"},
-		{"abc${AUCTION_AD_ID}def", bidRes, "abcSome ad id goes heredef"},
-		{"abc${AUCTION_PRICE}def", bidRes, "abc2.35def"},
-		{"abc${AUCTION_CURRENCY}def", bidRes, "abcUSDdef"},
+		{"abc${AUCTION_ID}def", "abc1234def"},
+		{"abc${AUCTION_IDD}def", "abc${AUCTION_IDD}def"},
+		{"abc${AUCTION_ID:B64}def", "abcMTIzNA==def"},
+		{"abc${AUCTION_BID_ID}def", "abcTheBidId!def"},
+		{"abc${AUCTION_IMP_ID}def", "abcImpressionIdForBiddef"},
+		{"abc${AUCTION_SEAT_ID}def", "abcSeatBidIdentifierdef"},
+		{"abc${AUCTION_AD_ID}def", "abcSome ad id goes heredef"},
+		{"abc${AUCTION_PRICE}def", "abc2.35def"},
+		{"abc${AUCTION_CURRENCY}def", "abcUSDdef"},
 		{
 			"${AUCTION_ID}${AUCTION_BID_ID}${AUCTION_IMP_ID}${AUCTION_SEAT_ID}" +
 				"${AUCTION_AD_ID}${AUCTION_AD_ID:B64}${AUCTION_PRICE}${AUCTION_CURRENCY}",
-			bidRes,
 			"1234TheBidId!ImpressionIdForBidSeatBidIdentifierSome ad id goes here" +
 				"U29tZSBhZCBpZCBnb2VzIGhlcmU=2.35USD",
 		},
-		{"abc${AUCTION_ID}${AUCTION_ID}def", bidRes, "abc12341234def"},
+		{"abc${AUCTION_ID}${AUCTION_ID}def", "abc12341234def"},
 	}
-
-	for i, tt := range macroSubsTests {
-		t.Logf("Testing %d", i)
-		verifyMacroSubs(t, tt.input, tt.bidRes, tt.expectedOutput)
+	for _, test := range tests {
+		actual, err := openrtb.MacroSubs(test.input, bidRes, ar)
+		if err != nil {
+			t.Errorf("MacroSubs err: %s", err)
+		}
+		if actual != test.expected {
+			t.Errorf("Expected \"%s\", but got: \"%s\"", test.expected, actual)
```

Comment:
Nitpick: I think it's more readable if we do string literal here to avoid additional process on the escape character:

```go
t.Errorf(`Expected "%s", but got: "%s"`, test.expected, actual)
```